commands/create_cluster.go

```diff
@@ -536,29 +527,14 @@ func addCluster(args addClusterArguments) (addClusterResult, error) {
 
 	if !args.dryRun {
 		fmt.Printf("Requesting new cluster for organization '%s'\n", color.CyanString(result.definition.Owner))
-
 		auxParams := ClientV2.DefaultAuxiliaryParams()
 		auxParams.ActivityName = createClusterActivityName
-
 		// perform API call
 		response, err := ClientV2.CreateCluster(addClusterBody, auxParams)
 		if err != nil {
-			// create specific error types for cases we care about
-			if clientErr, ok := err.(*clienterror.APIError); ok {
```

Comment:
Okay, I took my time and had a closer look at what is happening:
I assumed that `APIError` is a generated type, but it is actually an error type we define here: https://github.com/giantswarm/gsctl/blob/master/client/clienterror/clienterror.go
So the thing we effectively do here is compare an error type (`APIError`) which we create to another microerror error type (from https://github.com/giantswarm/gsctl/blob/master/commands/error.go) which we also create. 

The refactoring goal would be to move the commands to different packages and then only use the `microerror` type with proper matchers instead of defining everything twice.